lib_test/test_fs_unix.ml

```diff
@@ -106,9 +108,22 @@ let read_zero_bytes () =
       sure %s is present in the test filesystem" content_file)
   | `Error e -> assert_fail e
 
+(* reasonable responses are `Ok [] or an error *)
+let read_negative_bytes () =
+  connect_or_fail () >>= fun fs ->
+  FS_impl.read fs content_file 0 (-5) >>= function
+  | `Ok [] -> Lwt.return_unit
```

Comment:
If I remember correctly, I discovered this as part of a cascade of poor error-handling prompted by an off-by-some error in a Cstruct definition.  This happens when someone messed up at programming (in the prompting case, me), and I'm now more inclined to say that the result should be something louder and less easy to ignore.